hubspot/connection/__init__.py

```diff
@@ -204,6 +204,7 @@ def _require_successful_response(response):
             raise exception_class(
                 error_data['message'],
                 error_data['requestId'],
+                error_data,
```

Comment:
@frankoid That sounds good to me, I believe at the time we even thought about doing something like this even thought it does indeed break encapsulation, but I completely understand the pain of working with Hubspot's API and their documentation...

I'll discuss with my colleagues this specific part of your implementation, but in my opinion, that's great. The only thing that I'm thinking is whether to make the schema for `failureMessages` a little bit more complete, but without checking their API documentation it's hard to tell what should the schema be, but since the idea is to have flexibility with whatever `failureMessages` you get, I'd say that the `Optional('failureMessages'): list` should suffice